packages/rlc-common/src/buildSamples.ts

```diff
@@ -0,0 +1,835 @@
+  const values: string[] = [];
+  const schema = schemaMap.get(type) as ObjectSchema;
+  const allPropNames = new Set<string>();
+  extractObjectProperties(
+    schema?.properties ?? {},
+    schemaMap,
+    allPropNames,
+    values,
+    path
+  );
+
+  // add parents' properties
+  const parents = getAllParents(schema, schemaMap);
+  for (let parent of parents) {
+    extractObjectProperties(
+      (parent as ObjectSchema)?.properties ?? {},
+      schemaMap,
+      allPropNames,
+      values,
+      path
+    );
+  }
+
+  return `{${values.join(", ")}}`;
+}
+
+function getAllParents(
+  schema: ObjectSchema,
+  schemaMap: Map<string, Schema>
+): Schema[] {
+  if (!schema) {
+    return [];
+  }
+  const isVisited = new Set<string>();
+  dfs(schema);
+  return Array.from(isVisited)
+    .filter((p) => schemaMap.get(p) !== undefined)
+    .map((p) => schemaMap.get(p)!);
+
+  function dfs(root?: Schema) {
+    if (!root) {
+      return;
+    }
+    const parents = getImmediateParentsNames(root, [SchemaContext.Input]);
+    for (let parent of parents) {
+      if (isVisited.has(parent)) {
+        continue;
+      }
+      dfs(schemaMap.get(parent));
+      isVisited.add(parent);
+    }
+  }
+}
+
+function extractObjectProperties(
+  properties: Record<string, Schema>,
+  schemaMap: Map<string, Schema>,
+  allPropNames: Set<string>,
+  values: string[],
+  path: Set<string>
+) {
+  for (const prop in properties) {
+    const propName = prop;
+    const propertySchema = properties[prop];
+    if (propertySchema.readOnly) {
+      continue;
+    }
+    if (propertySchema.type === "never") {
+      continue;
+    }
+    if (allPropNames.has(propName)) {
+      continue;
+    }
+    const propType = propertySchema.typeName ?? propertySchema.type;
+    if (propType !== "string" && !schemaMap.has(propType)) {
+      schemaMap.set(propType, propertySchema);
+    }
+    allPropNames.add(propName);
+    let propRetValue =
+      `${propName}: ` +
+      mockParameterTypeValue(propType, propName, schemaMap, path);
+    values.push(propRetValue);
+  }
+}
+
+function isQuotedString(type: string) {
+  return (
+    /^"([a-zA-Z0-9=;\/\+\-\s]*)"$/g.test(type) ||
+    /^'([a-zA-Z0-9=;\/\+\-\s]*)'$/g.test(type)
+  );
+}
+
+function isRecord(type: string) {
+  return /^Record<([a-zA-Z].+),(\s*)([a-zA-Z].+)>$/g.test(type);
+}
+
+function getRecordType(type: string) {
+  return /Record<([a-zA-Z].+),(\s*)(?<type>[a-zA-Z].+)>/g.exec(type)?.groups
+    ?.type;
+}
+
+function isArray(type: string) {
+  return isArrayObject(type) || isNativeArray(type);
+}
+
+function isArrayObject(type: string) {
+  return /(^Array<([a-zA-Z].+)>$)/g.test(type);
+}
+
+function getArrayObjectType(type: string) {
+  return /Array<(?<type>[a-zA-Z].+)>/g.exec(type)?.groups?.type;
+}
+
+function isNativeArray(type: string) {
+  return /(^[a-zA-Z].+)\[\]$/g.test(type);
+}
+
+function getNativeArrayType(type: string) {
+  return /(?<type>[a-zA-Z].+)\[\]/g.exec(type)?.groups?.type;
+}
```

Comment:
Yes, I think it's okay but I may prefer to refactor the schema detail better if we have time. Because if we could have a clear definition we actually don't need to guess the type which means we don't guess the `number[]` is array.

How do you think?